src/manager.cpp

```diff
@@ -347,8 +347,100 @@ types::DbusVariantType
+        const auto l_dbusValue = dbusUtility::readDbusProperty(
+            constants::bmcStateService, constants::bmcZeroStateObject,
+            constants::bmcStateInterface, constants::currentBMCStateProperty);
+
+        if (const auto l_currentBMCSate =
+                std::get_if<std::string>(&l_dbusValue))
+        {
+            if (*l_currentBMCSate != constants::bmcReadyState)
+            {
+                throw std::runtime_error(
+                    "BMC not ready. Single FRU VPD collection failed for " +
+                    std::string(i_dbusObjPath));
+            }
+        }
+
+        // Get system config JSON object from worker class
+        nlohmann::json l_sysCfgJsonObj{};
+
+        if (m_worker.get() != nullptr)
+        {
+            l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+        }
+
+        // Check if system config JSON is present
+        if (l_sysCfgJsonObj.empty())
+        {
+            throw std::runtime_error(
+                "System config JSON object not present. Single FRU VPD collection failed for " +
+                std::string(i_dbusObjPath));
+        }
+
+        // Get FRU path for the given D-bus object path from JSON
+        const std::string& l_fruPath =
+            jsonUtility::getFruPathFromJson(l_sysCfgJsonObj, i_dbusObjPath);
+
+        if (l_fruPath.empty())
+        {
+            throw std::runtime_error(
+                "D-bus object path not present in JSON. Single FRU VPD collection failed for " +
+                std::string(i_dbusObjPath));
+        }
+
+        // Parse VPD
+        types::VPDMapVariant l_parsedVpd = m_worker->parseVpdFile(l_fruPath);
+
+        // If l_parsedVpd is pointing to std::monostate
+        if (l_parsedVpd.index() == 0)
+        {
+            throw std::runtime_error("VPD parsing failed for " +
+                                     std::string(i_dbusObjPath));
+        }
+
+        // Get D-bus object map from worker class
+        types::ObjectMap l_dbusObjectMap;
+        m_worker->populateDbus(l_parsedVpd, l_dbusObjectMap, l_fruPath);
+
+        if (l_dbusObjectMap.empty())
+        {
+            throw std::runtime_error(
+                "Failed to create D-bus object map. Single FRU VPD collection failed for " +
+                std::string(i_dbusObjPath));
+        }
+
+        // Call PIM's Notify method
+        if (!dbusUtility::callPIM(move(l_dbusObjectMap)))
+        {
+            throw std::runtime_error(
+                "Notify PIM failed. Single FRU VPD collection failed for " +
+                std::string(i_dbusObjPath));
+        }
+
+        // If PCA chip address is found in JSON, bind it.
+        const std::string& l_cmdToBindPCAChip =
+            jsonUtility::getPCAChipCmd(l_fruPath, l_sysCfgJsonObj, "bind");
+
+        if (!l_cmdToBindPCAChip.empty())
+        {
+            commonUtility::executeCmd(l_cmdToBindPCAChip);
+        }
+        else
+        {
+            throw std::runtime_error(
```

Comment:
Is it mandatory to have a PCA chip? 
Ideally this API should not take care of this. Collection process will take care of binding any PCA chip if required. (This is a TODO item.)

Reply:
yes, its not mandatory. since the "collection" tag in JSON doesn't have the command to bind pca chip, added the logic here.

So you mean, this is the TODO item to add pca bind cmd under "collection" tag in JSON?

Reply:
So if it not mandatory then the API will throw exception for all FRUs which does not have PCA. 
Anyways, this will be taken care in the collection path. This API need not take care of that.
Can you update the flow diagram as well.